api/src/main/java/org/pmiops/workbench/actionaudit/auditors/UserServiceAuditAdapterImpl.java

```diff
@@ -20,32 +21,50 @@
 @Service
 public class UserServiceAuditAdapterImpl implements UserServiceAuditor {
 
+  private static final Logger log = Logger.getLogger(UserServiceAuditAdapterImpl.class.getName());
+
   private final ActionAuditService actionAuditService;
   private final Clock clock;
-  private Provider<DbUser> dbUserProvider;
   private Provider<String> actionIdProvider;
+  private Provider<Optional<DbUser>> dbUserOptionalProvider;
 
   @Autowired
   public UserServiceAuditAdapterImpl(
       ActionAuditService actionAuditService,
       Clock clock,
-      Provider<DbUser> dbUserProvider,
-      @Qualifier("ACTION_ID") Provider<String> actionIdProvider) {
+      @Qualifier("ACTION_ID") Provider<String> actionIdProvider,
+      Provider<Optional<DbUser>> dbUserOptionalProvider) {
     this.actionAuditService = actionAuditService;
     this.clock = clock;
-    this.dbUserProvider = dbUserProvider;
     this.actionIdProvider = actionIdProvider;
+    this.dbUserOptionalProvider = dbUserOptionalProvider;
   }
 
   @Override
   public void fireUpdateDataAccessAction(
-      DbUser targetUser, DataAccessLevel dataAccessLevel, DataAccessLevel previousDataAccessLevel) {
+      DbUser targetUser,
+      DataAccessLevel dataAccessLevel,
+      DataAccessLevel previousDataAccessLevel,
+      AgentType agentType) {
+    // In the event of an offline cronjob, the request may have been made anonymously and there is
+    // no user identity. This should only happen for system agent driven audit events.
+    Optional<DbUser> agent = dbUserOptionalProvider.get();
+    if (agent.isPresent() && AgentType.SYSTEM == agentType) {
+      log.warning(
+          String.format(
+              "found unexpected request scope user '%s' for SYSTEM agent audit action",
+              agent.get().getUsername()));
+    } else if (!agent.isPresent() && AgentType.SYSTEM != agentType) {
+      log.warning(
+          String.format("missing request scope user for audit action agent type of %s", agentType));
+    }
+
```

Comment:
Can you please have a block for unhandled agent types? (A `switch` may be better actually)